ctx.go

```diff
@@ -911,6 +911,13 @@ func (c *Ctx) QueryParser(out interface{}) error {
 		k := utils.UnsafeString(key)
 		v := utils.UnsafeString(val)
 
+		if strings.Contains(k, "[") {
+			var err error
+			if k, err = parseQuery(k); err != nil {
+				panic(err)
```

Comment:
Queryparser has error return. I think you can return the error directly.
Also you don't need to `var err error` i suppose.

Reply:
No, I can't the code is inside `VisitAll()` and the function signature is `func(key, val []byte)` cannot return error directly from here, any other way you think so?

Reply:
Maybe you can create err var outside of VisitAll and then you can set and check it. But panic is nice way too. What do you think @ReneWerner87

Reply:
I could also do this:
```
func (c *Ctx) QueryParser(out interface{}) (err error) {
```
I can assign the returned error to the `err` variable in signature, wdyt?

Reply:
na, but the problem is it will star scanning the struct and I have to add a ugly error check after VisitAll, I think having error outside as a variable will be better

Reply:
```// QueryParser binds the query string to a struct.
func (c *Ctx) QueryParser(out interface{}) error {
	data := make(map[string][]string)
	var err error

	c.fasthttp.QueryArgs().VisitAll(func(key, val []byte) {
		k := utils.UnsafeString(key)
		v := utils.UnsafeString(val)

		if strings.Contains(k, "[") {
			k, err = parseQuery(k)
		}

		if strings.Contains(v, ",") && equalFieldType(out, reflect.Slice, k) {
			values := strings.Split(v, ",")
			for i := 0; i < len(values); i++ {
				data[k] = append(data[k], values[i])
			}
		} else {
			data[k] = append(data[k], v)
		}

	})

	if err != nil {
		return err
	}

	return c.parseToStruct(queryTag, out, data)
}
```

Reply:
> Na, but the problem is it will start scanning the struct and I have to add an ugly error check after `VisitAll`, I think having an error outside as a variable will be better

I'm thinking the same.

Reply:
2 different error handlings do not make sense here

either the method does panic on every error, which I would consider not good 

or the method returns the error as specified in the declaration

normally you would have to create a slice with errors and fill it in the run, because the run can not be aborted, otherwise the previous error might be overwritten by the next successful case

must unfortunately suggest something like this
```go
var err error

c.fasthttp.QueryArgs().VisitAll(func(key, val []byte) {
       if err != nil {
            return nil // skip the process for the next items
        }
        k := utils.UnsafeString(key)
        v := utils.UnsafeString(val)
        
        if strings.Contains(k, "[") {
	        k, err = parseQuery(k)
        }
        
        if strings.Contains(v, ",") && equalFieldType(out, reflect.Slice, k) {
	        values := strings.Split(v, ",")
	        for i := 0; i < len(values); i++ {
		        data[k] = append(data[k], values[i])
	        }
        } else {
	        data[k] = append(data[k], v)
        }

})

if err != nil {
    return err
}
```

Reply:
or you leave the key in the visitAll untouched and iterate over the data slice again after the method (which would probably be slower)

```go
var err error

c.fasthttp.QueryArgs().VisitAll(func(key, val []byte) {
        k := utils.UnsafeString(key)
        v := utils.UnsafeString(val)
               
        if strings.Contains(v, ",") && equalFieldType(out, reflect.Slice, k) {
	        values := strings.Split(v, ",")
	        for i := 0; i < len(values); i++ {
		        data[k] = append(data[k], values[i])
	        }
        } else {
	        data[k] = append(data[k], v)
        }

})
for k, v := range data {
        if strings.Contains(k, "[") {
                delete(data, k)
	        k, err := parseQuery(k)
	        if err != nil {
                      return err
                }
                data[k] = v
        }
}

```

Reply:
Changed to 1st version you @ReneWerner87 @efectn. If it looks good could you please approve the PR. Thank you 🎉